types/rdf-js/index.d.ts

```diff
@@ -476,3 +476,179 @@ export interface DatasetCoreFactory<Q extends BaseQuad = Quad> {
+
+    /**
+     * Executes the provided `iteratee` once on each quad in the dataset.
+     *
+     * This method is aligned with `Array.prototype.forEach()` in ECMAScript-262.
+     */
+    forEach(iteratee: QuadRunIteratee<Q>): void;
+
+    /**
+     * Imports all quads from the given stream into the dataset.
+     *
+     * The stream events `end` and `error` are wrapped in a Promise.
+     */
+    import(stream: Stream<Q>): Promise<Dataset<Q>>;
+
+    /**
+     * Returns a new dataset containing alls quads from the current dataset that are also included in the given dataset.
+     */
+    intersection(other: Dataset<Q>): Dataset<Q>;
+
+    /**
+     * Returns a new dataset containing all quads returned by applying `iteratee` to each quad in the current dataset.
+     */
+    map(iteratee: QuadMapIteratee<Q>): Dataset<Q>;
+
+    /**
+     * This method calls the `iteratee` on each `quad` of the `Dataset`. The first time the `iteratee` is called, the
+     * `accumulator` value is the `initialValue` or, if not given, equals to the first quad of the `Dataset`. The return
+     * value of the `iteratee` is used as `accumulator` value for the next calls.
+     *
+     * This method returns the return value of the last `iteratee` call.
+     *
+     * This method is aligned with `Array.prototype.reduce()` in ECMAScript-262.
+     */
+    reduce<A = any>(iteratee: QuadReduceIteratee<Q, A>, initialValue: A): A;
+
+    /**
+     * Existential quantification method, tests whether some quads in the dataset pass the test implemented by the
+     * provided `iteratee`.
+     *
+     * This method immediately returns boolean `true` once a quad that passes the test is found.
+     *
+     * This method is aligned with `Array.prototype.some()` in ECMAScript-262.
+     */
+    some(iteratee: QuadFilterIteratee<Q>): boolean;
+
+    /**
+     * Returns the set of quads within the dataset as a host language native sequence, for example an `Array` in
+     * ECMAScript-262.
+     *
+     * Since a `Dataset` is an unordered set, the order of the quads within the returned sequence is arbitrary.
+     */
+    toArray(): Q[];
+
+    /**
+     * Returns an N-Quads string representation of the dataset, preprocessed with
+     * {@link https://json-ld.github.io/normalization/spec/|RDF Dataset Normalization} algorithm.
+     */
+    toCanonical(): string;
+
+    /**
+     * Returns a stream that contains all quads of the dataset.
+     */
+    toStream(): Stream<Q>;
+
+    /**
+     * Returns an N-Quads string representation of the dataset.
+     *
+     * No prior normalization is required, therefore the results for the same quads may vary depending on the `Dataset`
+     * implementation.
+     */
+    toString(): string;
+
+    /**
+     * Returns a new `Dataset` that is a concatenation of this dataset and the quads given as an argument.
+     */
+    union(quads: Dataset<Q>): Dataset<Q>;
+}
+
+export interface DatasetFactory extends DataFactory {
```

Comment:
Is this meant to extend `DataFactory`? `DatasetCoreFactory` doesn't, and https://github.com/rdf-ext/rdf-dataset-indexed/blob/master/index.js doesn't implement it.

Reply:
That looks like a typo in the spec (feel free to open a PR for it). I think that should become `extends DatasetCoreFactory`.

Reply:
:+1: Makes more sense.

Reply:
https://github.com/rdfjs/dataset-spec/pull/57